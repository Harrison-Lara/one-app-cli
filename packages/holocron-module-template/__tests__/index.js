const templateApi = require('..');

describe('Template public api', () => {
  describe('getTemplatePaths', () => {
    it('should exist, and return an array of strings', () => {
      expect(typeof templateApi.getTemplatePaths).toBe('function');

      const paths = templateApi.getTemplatePaths();
      expect(Array.isArray(paths)).toBe(true);
      paths.forEach((path) => expect(typeof path).toBe('string'));
    });
  });
  describe('getTemplateOptions', () => {
    const baseData = {
      moduleName: 'module-name-mock',
    };
    let promptsMock;
    beforeEach(() => {
      promptsMock = jest.fn((arr) => arr.reduce((prompt, acc) => ({ ...acc, [prompt.name]: `${prompt.name}Mock` }), {}));
    });
    it('should exist and return an object with 1 required key', async () => {
      expect(typeof templateApi.getTemplateOptions).toBe('function');

      const templateOptions = await templateApi.getTemplateOptions(baseData, promptsMock);
      expect(typeof templateOptions).toBe('object');
      expect(typeof templateOptions.templateValues).toBe('object');
    });
    describe('templateOptions.templateValues', () => {
      it('should match the snapshot', async () => {
        const templateOptions = await templateApi.getTemplateOptions(baseData, promptsMock);
        expect(templateOptions.templateValues).toMatchSnapshot();
      });
      it('should match the snapshot when the prompts are all black', async () => {
        promptsMock = jest.fn((arr) => arr.reduce((prompt, acc) => ({ ...acc, [prompt.name]: '' }), {}));
        const templateOptions = await templateApi.getTemplateOptions(baseData, promptsMock);
        expect(templateOptions.templateValues).toMatchSnapshot();
      });
      it('should call Prompts with the correct config', async () => {
        await templateApi.getTemplateOptions(baseData, promptsMock);
        expect(promptsMock.mock.calls[0][0]).toMatchSnapshot();
      });
    });
    describe('templateOptions.dynamicFileNames', () => {
      it('should if present, should match the snapshot', async () => {
        const templateOptions = await templateApi.getTemplateOptions(baseData, promptsMock);
        expect(templateOptions.dynamicFileNames).toMatchSnapshot();
      });
    });
    describe('templateOptions.ignoredFileNames', () => {
      it('should if present, should match the snapshot', async () => {
        const templateOptions = await templateApi.getTemplateOptions(baseData, promptsMock);
        expect(templateOptions.ignoredFileNames).toMatchSnapshot();
      });
    });
  });
});
