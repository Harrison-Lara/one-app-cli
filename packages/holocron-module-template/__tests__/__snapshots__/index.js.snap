// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template public api getTemplateOptions templateOptions.dynamicFileNames should if present, should match the snapshot 1`] = `
Object {
  "RootComponent.jsx": "ModuleNameMock.jsx",
  "RootComponent.spec.js": "ModuleNameMock.spec.js",
  "RootComponent.spec.jsx": "ModuleNameMock.spec.jsx",
}
`;

exports[`Template public api getTemplateOptions templateOptions.ignoredFileNames should if present, should match the snapshot 1`] = `
Array [
  ".eslintrc.json",
  "appConfig.js",
  "appConfig.spec.js",
  "csp.js",
  "childRoutes.jsx",
]
`;

exports[`Template public api getTemplateOptions templateOptions.templateValues should call Prompts with the correct config 1`] = `
Array [
  Object {
    "initial": "",
    "message": "Enter module description:",
    "name": "description",
    "type": "text",
  },
  Object {
    "initial": "",
    "message": "Enter the name of your root module (leave blank for this module to be root):",
    "name": "rootModuleName",
    "type": "text",
  },
  Object {
    "initial": "",
    "message": "Enter the url for your dev environment module map:",
    "name": "moduleMapUrl",
    "type": "text",
  },
]
`;

exports[`Template public api getTemplateOptions templateOptions.templateValues should match the snapshot 1`] = `
Object {
  "camelCaseModuleName": "ModuleNameMock",
  "initial": "",
  "isRootModule": false,
  "message": "Enter the url for your dev environment module map:",
  "moduleName": "module-name-mock",
  "name": "moduleMapUrl",
  "rootModuleName": "rootModuleNameMock",
  "type": "text",
}
`;

exports[`Template public api getTemplateOptions templateOptions.templateValues should match the snapshot when the prompts are all black 1`] = `
Object {
  "camelCaseModuleName": "ModuleNameMock",
  "initial": "",
  "isRootModule": true,
  "message": "Enter the url for your dev environment module map:",
  "moduleName": "module-name-mock",
  "name": "moduleMapUrl",
  "rootModuleName": "module-name-mock",
  "type": "text",
}
`;
